storage/adapter.go

```diff
@@ -0,0 +1,407 @@
+
+	smsg, err := m.api.MpoolPushMessage(ctx, msg)
+	if err != nil {
+		return cid.Undef, xerrors.Errorf("pushing message to mpool: %w", err)
+	}
+
+	return smsg.Cid(), nil
+}
+
+func (m *StorageMinerNodeAdapter) WaitForProveCommitSector(ctx context.Context, proveCommitSectorMsgCid cid.Cid) (uint8, error) {
+	mw, err := m.api.StateWaitMsg(ctx, proveCommitSectorMsgCid)
+	if err != nil {
+		return 0, err
+	}
+
+	return mw.Receipt.ExitCode, nil
+}
+
+func (m *StorageMinerNodeAdapter) SendReportFaults(ctx context.Context, sectorIDs ...uint64) (cid.Cid, error) {
+	// TODO: check if the fault has already been reported, and that this sector is even valid
+
+	bf := types.NewBitField()
+	for _, id := range sectorIDs {
+		bf.Set(id)
+	}
+
+	enc, aerr := actors.SerializeParams(&actors.DeclareFaultsParams{bf})
+	if aerr != nil {
+		return cid.Undef, xerrors.Errorf("failed to serialize declare fault params: %w", aerr)
+	}
+
+	msg := &types.Message{
+		To:       m.maddr,
+		From:     m.waddr,
+		Method:   actors.MAMethods.DeclareFaults,
+		Params:   enc,
+		Value:    types.NewInt(0), // TODO: need to ensure sufficient collateral
+		GasLimit: types.NewInt(1000000 /* i dont know help */),
+		GasPrice: types.NewInt(1),
+	}
+
+	smsg, err := m.api.MpoolPushMessage(ctx, msg)
+	if err != nil {
+		return cid.Undef, xerrors.Errorf("failed to push declare faults message to network: %w", err)
+	}
+
+	return smsg.Cid(), nil
+}
+
+func (m *StorageMinerNodeAdapter) WaitForReportFaults(ctx context.Context, declareFaultsMsgCid cid.Cid) (uint8, error) {
+	mw, err := m.api.StateWaitMsg(ctx, declareFaultsMsgCid)
+	if err != nil {
+		return 0, err
+	}
+
+	return mw.Receipt.ExitCode, nil
+}
+
+func (m *StorageMinerNodeAdapter) GetSealTicket(ctx context.Context) (s2.SealTicket, error) {
+	ticket, err := m.tktFn(ctx)
+	if err != nil {
+		return s2.SealTicket{}, xerrors.Errorf("getting ticket failed: %w", err)
+	}
+
+	return s2.SealTicket{
+		BlockHeight: ticket.BlockHeight,
+		TicketBytes: ticket.TicketBytes[:],
+	}, nil
+}
+
+func (m *StorageMinerNodeAdapter) GetReplicaCommitmentByID(ctx context.Context, sectorID uint64) (commR []byte, wasFound bool, err error) {
+	act, err := m.api.StateGetActor(ctx, m.maddr, nil)
+	if err != nil {
+		return nil, true, err
+	}
+
+	st, err := m.api.ChainReadObj(ctx, act.Head)
+	if err != nil {
+		return nil, true, err
+	}
+
+	var state actors.StorageMinerActorState
+	if err := state.UnmarshalCBOR(bytes.NewReader(st)); err != nil {
+		return nil, true, xerrors.Errorf("unmarshaling miner state: %+v", err)
+	}
+
+	pci, found := state.PreCommittedSectors[fmt.Sprint(sectorID)]
+	if found {
+		// TODO: If not expired yet, we can just try reusing sealticket
+		return pci.Info.CommR, true, nil
+	}
+
+	return nil, false, nil
+}
+
+func (m *StorageMinerNodeAdapter) GetSealSeed(ctx context.Context, preCommitMsgCid cid.Cid, interval uint64) (<-chan s2.SealSeed, <-chan s2.SeedInvalidated, <-chan s2.FinalityReached, <-chan *s2.GetSealSeedError) {
```

Comment:
Instead of all the channels this should also just get a `statemachine.Context` and send the events directly